support style in link and map-link components in extensions

tbenr opened the pull request:
@jeluard @flexsurfer 

as discussed here: https://github.com/status-im/status-react/pull/7404#issuecomment-470143806

I implemented the ability to override the standard style for links.
I also added some new features to the link component:

`link  {:value link :properties {:uri :string :text? :string :open-in? {:one-of #{:device :status}}}}`

link definition now supports:
- displaying a `text` different from the `uri`
- with `open-in` you can choose to open the link on the device or in status browser. if you don't specify, dialog is shown to the user who choose how to open it.

extension example is here: https://get.status.im/extension/ipfs@QmXv1ZzHbJYiA7n8TM8aUaAQsW5Cm8jHj3MBmshVNP59qW


![image](https://user-images.githubusercontent.com/15999009/54025976-8e52b780-419c-11e9-85bd-8569f68e540c.png)

status-im-auto commented:
### Jenkins Builds
| :grey_question: | Commit | :hash: | Finished (UTC) | Duration | Platform | Result |
|-|-|-|-|-|-|-|
| :heavy_check_mark: | e34957db | [#2](https://ci.status.im/job/status-react/job/prs/job/macos/job/PR-7701/2/) | 2019-03-08 11:06:58 | ~16 min | `macos` | [:package: dmg](https://status-im-prs.ams3.digitaloceanspaces.com/StatusIm-190308-105014-e34957-pr.dmg) |

tbenr commented:
I added `[status-im.utils.http :as http]` but at the end i didn't used... 👎 
that's because i removed `http/normalize-url` from `#(.openURL react/linking uri)` because it doesn't support `geo:` android native URL protocol.

jeluard commented:
@tbenr Can you remove the `require` then? :)

tbenr committed:
support style in link

and map-link components in extensions

Signed-off-by: Julien Eluard <julien.eluard@gmail.com>

tbenr commented:
@jeluard done ;)
